tests/jax/integration/parameter_server_test.py

```diff
@@ -1,89 +1,243 @@
+
+    test_system_mt.launch()
+    time.sleep(10)
+
+    parameter_server = parameter_server_node._construct_instance()
+    executor = executor_node._construct_instance()
+    trainer = trainer_node._construct_instance()
+
+    # Initial state of the parameter_server
+    assert type(parameter_server.store.system_checkpointer) == savers.Checkpointer
+
+    for network in parameter_server.store.parameters["networks-network_agent"].values():
+        assert sorted(list(network.keys())) == ["b", "w"]
+        assert jnp.size(network["w"]) != 0
+        assert jnp.size(network["b"]) != 0
+
+    param_without_net = parameter_server.store.parameters.copy()
+    del param_without_net["networks-network_agent"]
+    assert param_without_net == {
+        "trainer_steps": jnp.zeros(1, dtype=jnp.int32),
+        "trainer_walltime": jnp.zeros(1, dtype=jnp.float32),
+        "evaluator_steps": jnp.zeros(1, dtype=jnp.int32),
+        "evaluator_episodes": jnp.zeros(1, dtype=jnp.int32),
+        "executor_episodes": jnp.zeros(1, dtype=jnp.int32),
+        "executor_steps": jnp.zeros(1, dtype=jnp.int32),
+    }
+
+    assert not parameter_server.store.last_checkpoint_time
+    assert parameter_server.store.system_checkpointer._last_saved == 0
+
+    # test get and set parameters
+    for _ in range(3):
+        executor.run_episode()
+    trainer.step()
+
+    trainer_steps = parameter_server.get_parameters("trainer_steps")
+    executor_episodes = parameter_server.get_parameters("executor_episodes")
+    assert list(trainer_steps) == [1]  # trainer.step one time
+    assert list(executor_episodes) == [3]  # run episodes three times
+
+    network = parameter_server.get_parameters("networks-network_agent")
+    for network in network.values():
+        assert sorted(list(network.keys())) == ["b", "w"]
+        assert jnp.size(network["w"]) != 0  # network is updated
+        assert jnp.size(network["b"]) != 0
+
+    # run step function
+    parameter_server.step()
+
+    assert parameter_server.store.last_checkpoint_time
+    assert parameter_server.store.last_checkpoint_time < time.time()
+    assert parameter_server.store.system_checkpointer._last_saved != 0
+    assert parameter_server.store.system_checkpointer._last_saved < time.time()
+
+
+def test_parameter_server_multi_process(test_system_mp: System) -> None:
+    """Test if the paraameter server instantiates processes as expected."""
+
+    (trainer_node,) = test_system_mp._builder.store.program._program._groups["trainer"]
+    (executor_node,) = test_system_mp._builder.store.program._program._groups[
+        "executor"
+    ]
+    (evaluator_node,) = test_system_mp._builder.store.program._program._groups[
+        "evaluator"
+    ]
+    (parameter_server_node,) = test_system_mp._builder.store.program._program._groups[
+        "parameter_server"
+    ]
+
+    trainer_node.disable_run()
+    executor_node.disable_run()
+    evaluator_node.disable_run()
+    # parameter_server_node.disable_run()
```

Comment:
Why don't we disable the parameter server as well?

Reply:
Yeah because I don't want to call explicitely the methods within the parameter server I wanted the executor and the trainer once they are running they will call set_parameter...

Reply:
also, the parameter server is disabled for multithread

Reply:
Oh I didn't notice I will try to disable it and see

Reply:
It works even when I desable the node I updated the code! Thanks for noticing